breadcrumbs.dust

```diff
@@ -0,0 +1,11 @@
+<ol class="c-breadcrumbs {class}" role="navigation" data-adaptivejs-component="stencil-breadcrumbs">
+    {?element}
+        {element|s}
+    {:else}
+        {#items}
+            <li class="c-breadcrumbs__item {?isActive}c--active{/isActive}">
+                {?href}<a href="{href}" class="c-breadcrumbs__link">{/href}{label}{?href}</a>{/href}
+            </li>
+        {/items}
+    {/element}
+</ol>
```

Comment:
This component is one (of many) that needs to render out a set of nearly identical items. It makes sense that a developer using the component should be able to write it out in shorthand form: `{@c-breadcrumbs items=items /}` especially since the data model for each item is so simple (just `href` and `label`).

However, as a general rule, *we should implement any repeating part of a component as its own template*. This allows for a developer using the component to have more control over how it renders. I’d recommend creating a sub-component template called `breadcrumbs__item.dust` and move the `<li>` into that. Then, update the main component template to look something like this:

```html
<ol class="c-breadcrumbs {class}" role="navigation" data-adaptivejs-component="stencil-breadcrumbs">
    {?body}
        {body}
    {:else}
        {#items}
            {@c-breadcrumbs__item ctx=. /}
        {/items}
    {/element}
</ol>
```

First, I’ve removed the ability to pass the entire component contents as a string `element`, for reasons noted above. Instead, I now check to see if there was a body provided, and if so, output it. If there is no body, I assume the user is using the shorthand and attempt to iterate over `items`. In that Dust section, I’m using the breadcrumbs__item sub-component template to render each item. 

Notice the special `ctx` parameter, to which we are passing `.`. This looks like a mistake, because Dust is weird, but that period represents the current context in its entirety (in this case the breadcrumbs item data). So what we are doing here is effectively passing all the keys (`href`, `label` and anything else) to the breadcrumbs__item template all in one go.

With these changes, here’s what I’d expect typical usage to look like:

```html
{! Expanded form, nothing fancy !}
{@c-breadcrumbs}
    {#items}
        {@c-breadcrumbs__item href=href label=label /}
    {#items}
{/c-breadcrumbs}
```

Notice that now the developer using this component has complete control over how the iteraction happens, and can even do things like pass different parameters based on conditions that the original component didn’t anticipate.

```html
{! Shorthand form, identical to before !}
{@c-breadcrumbs items=items /}
```

Finally, this is what the breadcrumbs item template might look like:

```html
<li class="c-breadcrumbs__item {class} {?isActive}c--active{/isActive}">
    {?href}
        <a href="{href}" class="c-breadcrumbs__link">{label}</a>
    {:else}
        <span class="c-breadcrumbs__link">{label}</span>
    {/href}
</li>
```

This allows for breadcrumbs items that are not links (e.g. the current item, since it links to the current page).